packages/core/admin/admin/src/content-manager/components/BlocksInput/utils/conversions.ts

```diff
@@ -89,4 +91,27 @@ const insertEmptyBlockAtLast = (editor: Editor) => {
   );
 };
 
-export { baseHandleConvert, getAttributesToClear, isLastBlockType, insertEmptyBlockAtLast };
+function useConversionModal() {
+  const [modalComponent, setModalComponent] = React.useState<React.JSX.Element | null>(null);
+
+  const handleConversionResult = (
+    maybeRenderModal: void | (() => React.JSX.Element) | undefined
+  ) => {
+    // Not all blocks return a modal
+    if (maybeRenderModal) {
+      // Use cloneElement to apply a key because to create a new instance of the component
+      // Without the new key, the state is kept from previous times that option was picked
+      setModalComponent(React.cloneElement(maybeRenderModal(), { key: Date.now() }));
+    }
+  };
+
+  return { modalComponent, handleConversionResult };
```

Comment:
If it's a component, should it be pascal case? ModalComponent

Reply:
I just tried to make it a component again for a bit, but still with no luck. I renamed it to modalElement so that it can at least be more clear about what it is

Reply:
That works thanks for taking a look, I'll try to QA both your PRs tonight or first thing in the morning to give you the ✅